cylc/flow/ws_data_mgr.py

```diff
@@ -364,9 +422,12 @@ def update_task_proxies(self, task_ids=None):
                 prereq_obj = prereq.api_dump(self.workflow_id)
                 if prereq_obj:
                     prereq_list.append(prereq_obj)
+            del tproxy.prerequisites[:]
             tproxy.prerequisites.extend(prereq_list)
```

Comment:
Any reason for maintaining the same list?

```diff
-            del tproxy.prerequisites[:]
-            tproxy.prerequisites.extend(prereq_list)
+            tproxy.prerequisites = prereq_list
```

Reply:
Can't do that with [repeated message fields](https://developers.google.com/protocol-buffers/docs/reference/python-generated#repeated-message-fields)